control-plane/grpc/proto/v1/frontend_node/registration.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "v1/misc/common.proto";
+
+package v1.frontend_node.registration;
+
+// Register the frontend node with control-plane.
+message RegisterRequest {
+  // Id of the frontend node instance.
```

Comment:
Could this just be the spec?

Reply:
I don't think we should do that, it's the same reason we have different types for NodeSpec and Register. If we add more fields to NodeSpec that doesn't need to be a part of registration.

Reply:
Right, good point.
In Proto is there any way to do type aliasing?

Reply:
Unfortunately no, although what we can do is wrap a message inside another, but again that creates an optional field inside.